### PR TITLE
feat(core): allow controllers to be bound via constructor

### PR DESCRIPTION
**Pull request by virkt25**
An array of controllers can now be passed into the array to be bound
automatically.

### CODE REVIEW

File: packages/core/src/application.ts [hunk 1]
```diff
@@ -30,6 +30,12 @@ export class Application extends Context {
         this.server(options.servers[name], name);
       }
     }
+
+    if (options.controllers) {
+      for (const controller in options.controllers) {
+        this.controller(controller);
```
Comment (raymondfeng) [inline packages/core/src/application.ts]: This is wrong. It should be `this.controller(options.controllers[controller])`. The current code fails CI as follows:
```
packages/core/src/application.ts(36,25): error TS2345: Argument of type 'string' is not assignable to parameter of type 'Constructor<any>'.
```

Comment (raymondfeng) [reply packages/core/src/application.ts]: If `options.controllers` is an array, `for...in` loop should not be used. Use 'for...of` instead.

**Comment by raymondfeng** [comment]
This is strong hint that an application itself is a component or it contains a default component. The `options.*` is the config for the component.

**Comment by virkt25** [comment]
@raymondfeng Fixed the code. Didn't realize I had used the `for` incorrectly. 

> This is strong hint that an application itself is a component or it contains a default component. The options.* is the config for the component.

Unfortunately I'm not sure what you mean by your overall comment. Can you please help me understand. Thanks.

**Comment by raymondfeng** [comment]
> Unfortunately I'm not sure what you mean by your overall comment. Can you please help me understand. Thanks.

@virkt25 : If you look at the configuration options being added to the application, there are common ones for an application and a component. If we can make application itself a component or embed a default component with it, the code will be cleaner.

**Comment by raymondfeng** [comment]
Let's merge the PR as is. We'll defer the application/component alignment.

**Pull request merged**